controller/src/beerocks/master/db/db.cpp

```diff
@@ -2423,73 +2423,257 @@ const std::list<sChannelScanResults> &db::get_channel_scan_results(const sMacAdd
 //
 std::chrono::steady_clock::time_point db::get_client_persistent_data_last_edit(const sMacAddr &mac)
 {
-    return std::chrono::steady_clock::time_point::min();
+    auto n = get_node(mac);
+    if (!n) {
+        LOG(ERROR) << "node not found.... ";
+        return std::chrono::steady_clock::time_point::min();
+    } else if (n->get_type() != beerocks::TYPE_CLIENT) {
+        LOG(ERROR) << __FUNCTION__ << "node " << mac << " is not a valid client!";
+        return std::chrono::steady_clock::time_point::min();
+    }
+
+    return n->client_persistent_data_last_edit;
 }
 
 bool db::set_client_time_life_delay(const sMacAddr &mac,
                                     const std::chrono::seconds &time_life_delay_sec,
                                     bool set_persistent_db)
 {
+    auto n = get_node(mac);
+    if (!n) {
+        LOG(ERROR) << "node not found.... ";
+        return false;
+    } else if (n->get_type() != beerocks::TYPE_CLIENT) {
+        LOG(ERROR) << __FUNCTION__ << "node " << tlvf::mac_to_string(mac)
+                   << " is not a valid client!";
+        return false;
+    }
+
+    LOG(DEBUG) << __FUNCTION__ << ", time_life_delay_sec = " << time_life_delay_sec.count();
+
+    auto timestamp = std::chrono::steady_clock::now();
+    if (set_persistent_db) {
+        LOG(DEBUG) << __FUNCTION__ << ", configuring persistent-db, time_life_delay_sec = "
+                   << time_life_delay_sec.count();
+    }
+
+    n->client_time_life_delay_sec       = time_life_delay_sec;
+    n->client_persistent_data_last_edit = timestamp;
+
     return true;
 }
 
 std::chrono::seconds db::get_client_time_life_delay(const sMacAddr &mac)
 {
-    return std::chrono::seconds(0);
+    auto n = get_node(mac);
+    if (!n) {
+        LOG(ERROR) << "node not found.... ";
+        return std::chrono::seconds::zero();
+    } else if (n->get_type() != beerocks::TYPE_CLIENT) {
+        LOG(ERROR) << __FUNCTION__ << "node " << mac << " is not a valid client!";
+        return std::chrono::seconds::zero();
+    }
+
+    return n->client_time_life_delay_sec;
 }
 
 bool db::set_client_stay_on_initial_radio(const sMacAddr &mac, bool stay_on_initial_radio,
                                           bool set_persistent_db)
 {
+    auto n = get_node(mac);
+    if (!n) {
+        LOG(ERROR) << "node not found.... ";
+        return false;
+    } else if (n->get_type() != beerocks::TYPE_CLIENT) {
+        LOG(ERROR) << __FUNCTION__ << "node " << mac << " is not a valid client!";
+        return false;
+    }
+
+    auto timestamp = std::chrono::steady_clock::now();
+    if (set_persistent_db) {
+        LOG(DEBUG) << __FUNCTION__
+                   << ", configuring persistent-db, stay_on_initial_radio = " << std::boolalpha
```

Comment:
boolalpha is configured by default on beerocks logging. No need to add it.

Reply:
reomved it